stellargraph/mapper/sampled_link_generators.py

```diff
@@ -216,9 +217,8 @@ def __init__(self, G, batch_size, num_samples, seed=None, name=None):
         self.head_node_types = self.schema.node_types * 2
 
         self._graph = G
-        self._seed = seed
-        self.random = random.Random(seed)
-        self._samplers = dict()
+        self._rs, _ = random_state(seed)
```

Comment:
Maybe this could be named to suggest that it should only be used for creating the batch things, e.g.:

```suggestion
        self._batch_sampler_rs, _ = random_state(seed)
```